### PR TITLE
Data Browser Charter

### PR DESCRIPTION
**Pull request by theathorn**
~~**October 19**: Last call for community reviews~~
~~**October 26**: Last call for oversight reviews~~
**November 15**: Approved during DCP Architecture call

### CODE REVIEW

File: charters/DataBrowser/charter.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+### Community engagement
+* Provide triage of feature requests from the community and integration of such requests into the Data Browser roadmap.
+* Outreach and engagement of the community on use/usability of the APIs.
+
+## Out-of-scope
+* Implementation of other language bindings for the APIs.
+
+## Milestones and Deliverables
+* Nov 2018:  Deploy a minimum viable product (MVP) feature set as part of the HCA DCP beta test phase.
+* Q4  2018:  Interface to the Matrix Service.
```
Comment (brianraymor) [inline charters/DataBrowser/charter.md]: Matrix Service -> **Expression** Matrix Service
Based on the current plan of record, isn't the interface to the expression matrix service included in the Nov 2018 milestone?

File: charters/DataBrowser/charter.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+### Project Lead
+[Brian O’Connor](mailto:brocono@ucsc.edu) 
+
+### Product Owner
+[Trevor Heathorn](mailto:theathor@ucsc.edu) 
+
+### Technical Lead
+[Hannes Schmidt](mailto:hannes@ucsc.edu)
+
+## Communication
```
Comment (brianraymor) [inline charters/DataBrowser/charter.md:59]: Is there a google group mailing list?

File: charters/DataBrowser/charter.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Definitions
+*DCP* The Data Coordination Platform is the name given to the entire system used to ingest, validate, store, analyzes, and make available the data in the Human Cell Atlas project.
+
+*Data Portal* The landing site for the Human Cell Atlas (HCA) DCP.
+
+*Analysis Applications* Third party portals listed in the 'Analyze' section of the Data Portal UI.
+
+## Objectives
+The objective of the Data Browser is to provide an easy-to-use User Interface (UI) allowing users to search, download and analyze data stored in the DCP at petabyte scale on public clouds.
```
Comment (tburdett) [inline charters/DataBrowser/charter.md]: This suggests just user interface, whereas the core capabilities describe an API too. Update this to resolve the conflict (I think you probably want to add API here?)

File: charters/DataBrowser/charter.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+The Data Browser (available via the Data Portal 'Explore' section) is an application that provides a faceted search capability allowing users to search and download subsets of cellular resolution data hosted by the Data Coordination Platform (DCP).
+
+## Definitions
+*DCP* The Data Coordination Platform is the name given to the entire system used to ingest, validate, store, analyzes, and make available the data in the Human Cell Atlas project.
+
+*Data Portal* The landing site for the Human Cell Atlas (HCA) DCP.
+
+*Analysis Applications* Third party portals listed in the 'Analyze' section of the Data Portal UI.
+
+## Objectives
```
Comment (tburdett) [inline charters/DataBrowser/charter.md:14]: I'll ref [my comment](https://github.com/HumanCellAtlas/dcp-community/pull/10#discussion_r223229320) on the portal charter here.

I want to see an indication somewhere of who is responsible for providing users of data from the DCP a sensible view over that data. Over time, metadata and data will change - in terms of schema, data file formats, QC, analysis results generated from new versions of pipelines. Where in the DCP is this wrangled into a single up-to-date view to provide some guarantees to users over the quality and contemporaneity of the data they get from us?

**Review by tburdett — CHANGES_REQUESTED** [review]
This is mostly clear. I have one UI/API comment that needs to be addressed, otherwise I am happy to approve. I've asked one very big question that needs answering _somewhere_ (but not necessarily here)

File: charters/DataBrowser/charter.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# Data Browser
+
+
+## Description
+The Data Browser (available via the Data Portal 'Explore' section) is an application that provides a faceted search capability allowing users to search and download subsets of cellular resolution data hosted by the Data Coordination Platform (DCP).
+
+## Definitions
+*DCP* The Data Coordination Platform is the name given to the entire system used to ingest, validate, store, analyzes, and make available the data in the Human Cell Atlas project.
```
Comment (malloryfreeberg) [inline charters/DataBrowser/charter.md]: Replace "analyzes" with "analyze"

File: charters/DataBrowser/charter.md (hunk 1 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md]: Agreed.

File: charters/DataBrowser/charter.md (hunk 2 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md:59]: Not at the moment. May add later.

File: charters/DataBrowser/charter.md (hunk 3 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md]: Agreed.

File: charters/DataBrowser/charter.md (hunk 4 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md:14]: A larger DCP-wide discussion outside the remit of just this charter.

File: charters/DataBrowser/charter.md (hunk 5 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md]: Done.

File: charters/DataBrowser/charter.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+### Technical Lead
+[Hannes Schmidt](mailto:hannes@ucsc.edu)
+
+## Communication
+
+### Slack Channels
+* HumanCellAtlas/data-browser: data browser development discussions
+
+## Github repositories
```
Comment (brianraymor) [inline charters/DataBrowser/charter.md:64]: Could you add a brief description of the repo(s)?

Comment (theathorn) [reply charters/DataBrowser/charter.md:64]: Done.

File: charters/DataBrowser/charter.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+# Data Browser
+
+
+## Description
+The Data Browser (available via the Data Portal 'Explore' section) is an application that provides a faceted search capability allowing users to search and download subsets of cellular resolution data hosted by the Data Coordination Platform (DCP).
+
```
Comment (lauraclarke) [inline charters/DataBrowser/charter.md:6]: Should this mention the goal of acting like a shopping cart for analysis applications? that isn't evident from this description?

File: charters/DataBrowser/charter.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+*Analysis Applications* Third party portals listed in the 'Analyze' section of the Data Portal UI.
+
+## Objectives
+The objective of the Data Browser is to provide an easy-to-use User Interface (UI) and RESTful API allowing users to search, download and analyze data stored in the DCP at petabyte scale on public clouds.
+
+## In-scope
+
+### Core capabilities
+* Create indexes to access files in the Data Store by using public Data Store Service (DSS) APIs to read the metadata.
```
Comment (lauraclarke) [inline charters/DataBrowser/charter.md:20]: What is the relationship between the Data Browser and the Query service?

File: charters/DataBrowser/charter.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+* Support multiple metadata schemas via a shared schema access library.
+* Provide multiple search views of the DCP data such as by Projects, Files and Specimens.
+* Support for specifying multiple search criteria ('facets') through the UI.
+* Provide for download of selected files from the Data Store via the UI and/or HCA DCP Command Line Interface (CLI).
```
Comment (lauraclarke) [inline charters/DataBrowser/charter.md]: When you use the word selected here do you mean user selected or that the browser will only enable download of a subset of files?

File: charters/DataBrowser/charter.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+* API documentation for portable, cloud-neutral RESTful APIs for faceted search of DCP data.
+
+### Security
+* Leverage the authentication and authorization implementation libraries from the DevSecOps team to implement user authentication and access control for the Data Browser.
+* Implement and configure tools to facilitate the operation of the Data Browser in a production environment.
+* Provide day-to-day operation of the Data Browser production deployment including but not limited to monitoring of errors and health of intrinsic components, fielding help requests, monitoring security, and assisting with the roll-out of new data into the deployment.
+
+### Community engagement
+* Provide triage of feature requests from the community and integration of such requests into the Data Browser roadmap.
+* Outreach and engagement of the community on use/usability of the APIs.
```
Comment (lauraclarke) [inline charters/DataBrowser/charter.md]: Just the APIs or also the UI?

**Review by lauraclarke — CHANGES_REQUESTED** [review]

File: charters/DataBrowser/charter.md (hunk 7 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md:6]: Added such a clause to the Description.

File: charters/DataBrowser/charter.md (hunk 8 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md:20]: The Data Browser receives notifications from the Data Store and constructs its own Indexes. It does not use the Query Service.

File: charters/DataBrowser/charter.md (hunk 9 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md]: Changed to "user-selected".

File: charters/DataBrowser/charter.md (hunk 10 above)
Comment (theathorn) [reply charters/DataBrowser/charter.md]: Added "UI".

**Review by tburdett — APPROVED** [review]
Thanks for the changes, this looks good

**Review by lauraclarke — APPROVED** [review]

**Pull request merged**